content/posts/numpy/numpy-rng/index.md

```diff
@@ -46,7 +46,7 @@ rng = np.random.default_rng()
 rng.random()  # generate a floating point number between 0 and 1
 ```
 
-If you want to use a seed for reproducibility, [the NumPy documentation recommends](https://numpy.org/doc/stable/reference/random/index.html#quick-start) using a very large and unique number to have a different seed than anyone else and thus make sure you will have independent results. To obtain a good and unique seed you can rely on the [secrets module](https://docs.python.org/3/library/secrets.html). The following code will return a 128 bit number that you can then use as a seed:
+If you want to use a seed for reproducibility, [the NumPy documentation](https://numpy.org/doc/stable/reference/random/index.html#quick-start) recommends using a large random number. The first reason is that this increases the probability of having a different seed than anyone else and thus make sure you will have independent results. The second reason is that using only small numbers for your seeds can result into biases as they will not cover the state space of the RNG. You should use at least 128-bit numbers (check [this comment](https://github.com/numpy/numpy/issues/25778#issuecomment-1930441151) by Robert Kern for more information)[^1]. To generate a 128-bit random number for your seed you can rely on the [secrets module](https://docs.python.org/3/library/secrets.html):
```

Comment:
```suggestion
If you want to use a seed for reproducibility, [the NumPy documentation](https://numpy.org/doc/stable/reference/random/index.html#quick-start) recommends using a large random number. The first reason is that this increases the probability of having a different seed than anyone else and thus make sure you will have independent results. The second reason is that using only small numbers for your seeds can result into biases as they will not cover the state space of the RNG. You should use at least 128-bit numbers (check [this comment](https://github.com/numpy/numpy/issues/25778#issuecomment-1930441151) by Robert Kern for more information)[^1]. To generate a 128-bit random number for your seed you can rely on the [secrets module](https://docs.python.org/3/library/secrets.html):
```

What do you mean they will not cover the state space of the PRNG? Presumably, numbers generated from a small seed has just as much a chance of accomplishing that as from large seeds.

Reply:
Yes I thought the same at first :) It's not about the sequence of numbers you will get with one seed but about the first number you will generate with your RNG after it has been seeded. If you only seed your RNG with small seeds (smaller than the state space), and you generate an integer from it, there are for instance numbers that you will never get as the first number thus creating a bias. https://www.pcg-random.org/posts/cpp-seeding-surprises.html#bias

Reply:
I could add a small explanation. or maybe the link with Robert Kern's comment appears a bit too far.

Reply:
I added some details here.